v1/plugin/plugin.go

```diff
@@ -328,5 +384,186 @@ func startPlugin(srv server, m *meta, p *pluginProxy) int {
+	cPolicy, err := p.plugin.(Collector).GetConfigPolicy()
+	if err != nil {
+		return err
+	}
+	fmt.Println("Config Policy:")
+	if len(cPolicy.stringRules) > 0 {
+		for k, v := range cPolicy.stringRules {
+			fmt.Printf("    Namespace: %-30v", k)
+			for key, val := range v.Rules {
+				fmt.Printf("  Key: %-10v  Info: ", key)
+				if val.String() != "" {
+					_, ok := conf[key]
+					if strings.Contains(val.String(), "required:true") && !ok {
+						requiredConfigs += "! Warning: \"" + key + "\" required by plugin and not provided in config \n"
+					}
+					fmt.Printf("%v", val.String())
+					//TODO: Check if config values are of correct type
+				}
+			}
+			fmt.Printf("\n")
+		}
+	}
+	if len(cPolicy.integerRules) > 0 {
+		for k, v := range cPolicy.integerRules {
+			fmt.Printf("    Namespace: %-30v", k)
+			for key, val := range v.Rules {
+				fmt.Printf("  Key: %-10v  Info: ", key)
+				if val.String() != "" {
+					_, ok := conf[key]
+					if strings.Contains(val.String(), "required:true") && !ok {
+						requiredConfigs += "! Warning: \"" + key + "\" required by plugin and not provided in config \n"
+					}
+					fmt.Printf("%v", val.String())
+				}
+			}
+			fmt.Printf("\n")
+		}
+	}
+	if len(cPolicy.floatRules) > 0 {
+		for k, v := range cPolicy.floatRules {
+			fmt.Printf("    Namespace: %-30v", k)
+			for key, val := range v.Rules {
+				fmt.Printf("  Key: %-10v  Info: ", key)
+				if val.String() != "" {
+					_, ok := conf[key]
+					if strings.Contains(val.String(), "required:true") && !ok {
+						requiredConfigs += "! Warning: \"" + key + "\" required by plugin and not provided in config \n"
+					}
+					fmt.Printf("%v", val.String())
+				}
+			}
+			fmt.Printf("\n")
+		}
+	}
+	if len(cPolicy.boolRules) > 0 {
+		for k, v := range cPolicy.boolRules {
+			fmt.Printf("    Namespace: %-30v", k)
+			for key, val := range v.Rules {
+				fmt.Printf("  Key: %-10v  Info: ", key)
+				if val.String() != "" {
+					_, ok := conf[key]
+					if strings.Contains(val.String(), "required:true") && !ok {
+						requiredConfigs += "! Warning: \"" + key + "\" required by plugin and not provided in config \n"
+					}
+					fmt.Printf("%v", val.String())
+				}
+			}
+			fmt.Printf("\n")
+		}
+	}
+
+	if requiredConfigs != "" {
+		requiredConfigs += "! Please provide config in form of: -config '{\"key\":\"kelly\", \"spirit-animal\":\"coatimundi\"}'\n"
+		err := fmt.Errorf(requiredConfigs)
+		return err
+	}
+
+	return nil
+}
+
+func printCollectMetrics(p *pluginProxy, m []Metric) error {
+	defer timeTrack(time.Now(), "printCollectMetrics")
+	cltd, err := p.plugin.(Collector).CollectMetrics(m)
+	if err != nil {
+		return fmt.Errorf("! Error in the call to CollectMetrics. Please ensure your config contains any required fields mentioned in the error below. \n %v", err)
+	}
+	fmt.Println("Metrics that can be collected right now are: ")
+	for _, j := range cltd {
+		fmt.Printf("    Namespace: %-30v  Type: %-10T  Value: %v \n", j.Namespace, j.Data, j.Data)
```

Comment:
I meant to say that the metric namespace seems good enough here. There is no need to print out the metric data. Am I missing anything?